src/Microsoft.VisualStudio.Editors/PropPages/PackagePropPage.vb

```diff
@@ -388,26 +387,19 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
 
         Private Sub AddOrChangeLicenseItem(oldInclude As String, newInclude As String)
             Dim projectLock = _configuredProject.Services.ExportProvider.GetExportedValue(Of IProjectLockService)()
-#Disable Warning RS0030 ' Do not used banned APIs
+#Disable Warning RS0030 ' Do not used banned APIs. The project lock is needed here - there is no IVT for ProjectAccessor
             ThreadHelper.JoinableTaskFactory.Run(
                 Async Function()
                     Await projectLock.WriteLockAsync(
                         Async Function(access)
                             Await access.CheckoutAsync(_unconfiguredProject.FullPath)
                             Dim projectXML = Await access.GetProjectXmlAsync(_unconfiguredProject.FullPath)
-                            Dim foundAndUpdated = False
-                            For Each itemGroup In projectXML.ItemGroups
-                                For Each item In itemGroup.Items
-                                    If oldInclude = item.Include Then
-                                        item.Include = newInclude
-                                        foundAndUpdated = True
-                                        Exit For
-                                    End If
-                                Next
-                            Next
-                            If (Not foundAndUpdated) Then
+                            Dim foundItem = projectXML.ItemGroups.SelectMany(Function(x) x.Items).FirstOrDefault(Function(x) x.Include = oldInclude)
+                            If foundItem IsNot Nothing Then
+                                foundItem.Include = newInclude
+                            Else
                                 'We couldn't find one to change so we should add it 
-                                projectXML.AddItem("None", newInclude, {(New KeyValuePair(Of String, String)("Pack", "True")), New KeyValuePair(Of String, String)("PackagePath", "")})
+                                projectXML.AddItem("None", newInclude, {New KeyValuePair(Of String, String)("Pack", "True"), New KeyValuePair(Of String, String)("PackagePath", "")})
```

Comment:
This logic of modifying the project XML does not belong in the UI layer. If we make changes to how things are written to the project file (for example, when we changed from elements -> attributes) this will be affected. 

The relationship here is that the UI sets/reads properties from the project system via property APIs, and the project system is responsible for mapping the XML => how its represented in-memory. I get this late now, but post Preview 4, please move this logic behind a project property (in a rule file with a datasource) and have that property read/write the item.

Reply:
Understood. I will get a PR out to update this behavior.